index.js

```diff
@@ -16,14 +19,20 @@ class Field {
    * @param {boolean} hardMode When true, a hole is randomly added after 5
    * turns.
    */
-  constructor(field, hardMode) {
+  constructor(field, hardMode = false) {
     this._field = field;
     this._hardMode = hardMode;
-    this._x = 0;
-    this._y = 0;
     this._turnCount = 0;
     this._height = field.length;
     this._width = field[0].length;
+
+    const startNode = Field.findNodeForTile(
+        field,
+        this._height,
+        this._width,
+        pathCharacter);
+    this._x = startNode.col;
+    this._y = startNode.row;
```

Comment:
The different naming convention used here for rows and columns is confusing. 

Consider renaming all variables to either `row` or `col`.